Assets/Virtual Agents Framework/Runtime/Scripts/ScheduleBasedExecution/TaskActions.cs

```diff
@@ -119,16 +120,23 @@ public AgentBaseTask PickUp(GameObject pickupObject, int priority = 0, SocketId
         /// </summary>
         /// <param name="destinationObject">Object the agent should go to and pick up. Needs to have an item component and be reachable by the agent.</param>
         /// <param name="priority">Priority of the task. Tasks with high importance should get a positive value, less important tasks a negative value. Default tasks have a priority of 0.</param>
-        /// <param name="bodyAttachPoint">Bodypart that the object should be attached to, standard is the right Hand</param>
+        /// <param name="bodyAttachPoint">Agent socket that the object should be attached to, standard is the right Hand</param>
         /// <param name="minDistance">Distance at which the the agent will try to pick up the object</param>
         /// <returns></returns>
         public AgentBaseTask GoToAndPickUp(GameObject destinationObject, int priority = 0, SocketId bodyAttachPoint = SocketId.RightHand, float minDistance = 0.3f)
         {
-            AgentMovementTask goToTask = (AgentMovementTask)GoTo(destinationObject, default, priority, true);
-            goToTask.MinDistance = minDistance;
+            List<AgentBaseTask> tasks = new List<AgentBaseTask>();
 
-            AgentBaseTask pickUpTask = PickUp(destinationObject, priority, bodyAttachPoint);
-            return pickUpTask;
+            AgentMovementTask movementTask = new AgentMovementTask(destinationObject, default, true);
+            movementTask.MinDistance = minDistance;
+            tasks.Add(movementTask);
+
+            AgentPickUpTask pickUpTask = new AgentPickUpTask(destinationObject, bodyAttachPoint);
+            tasks.Add(pickUpTask);
+
+            TaskBundle PickUpBundle = new TaskBundle(tasks);
```

Comment:
It might be a matter of preference but it could be more intuitive if the task bundle can be declared first and then, tasks are added to it one after another, instead of collecting them in a list first and then handing over the list. Something like this:

```
TaskBundle pickUpBundle = new TaskBundle();
AgentMovementTask movementTask = new AgentMovementTask(destinationObject, default, true);
movementTask.MinDistance = minDistance;
pickUpBundle.AddTask(movementTask);
pickUpBundle.AddTask(new AgentPickupTask(destinationObject, bodyAttachPoint));
```

We could then also allow a similar thing for the precoditions, so that they can also be added one after another.

Reply:
Both methods are now possible with a warning that preconditions that are added after the bundle started executing will not be checked. [4643371](https://github.com/rwth-acis/Virtual-Agents-Framework/pull/17/commits/4643371aa604a2038bf9a219e5028578a832af79)
The Sample Scene also shows both ways now.
Should I change the TaskActions to only use the AddTask and AddPrecondition methods, or leave them like before?

Reply:
Great! I think the TaskActions can stay as they are since their main purpose is to construct the task quickly.